### ISSUE TITLE
NVMe device with vfio-pci driver will occur error DMA message

### ISSUE THREAD

**Issue by MengjinWu** [spdk/spdk#3193]
# Sighting report

Attach a NVMe device with vfio-pci driver to spdk target will occur error messages when target exit.

## Expected Behavior

No error messages.

## Current Behavior

After I close the target, error messages occur

```
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
EAL: Unexpected size 0 of DMA remapping cleared instead of 2097152
```

## Steps to Reproduce

1. Ensure that NVMe driver binding with vfio-pci driver
2. Start the target
3. Attach NVMe Devices
4. Close the target

## Context (Environment including OS version, SPDK version, etc.)

Linux version 5.15.103.bsk.1-amd64 (gcc (Debian 8.3.0-6) 8.3.0, GNU ld (GNU Binutils for Debian) 2.31.1) 
SPDK version origin/v23.01.x

**Comment by jimharris** [comment]
@changpe1, can you give some guidance on this one?  It is same as issue #2075 which was previously closed. Were there fixes that went in after v23.01?

@MengjinWu do you also see this problem with SPDK master?

**Comment by changpe1** [comment]
> @changpe1, can you give some guidance on this one? It is same as issue #2075 which was previously closed. Were there fixes that went in after v23.01?
> 
> @MengjinWu do you also see this problem with SPDK master?

Yes, DPDK already fixed this issue, SPDK master doesn't have this issue.
